ccx_messaging/publishers/idp_rule_processing_publisher.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2019, 2020, 2021, 2022 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module that implements a custom Kafka publisher."""
+
+import json
+import logging
+from typing import Any
+
+import jsonschema
+
+from ccx_messaging.error import CCXMessagingError
+from ccx_messaging.publishers.kafka_publisher import KafkaPublisher
+from ccx_messaging.schemas import ARCHIVE_SYNCED_SCHEMA
+
+
+log = logging.getLogger(__name__)
+
+
+
+class IDPRuleProcessingPublisher(KafkaPublisher):
+    """RuleProcessingPublisher handles the results of the applied rules and publish them to Kafka.
+
+    The results of the data analysis are received as a JSON (string)
+    and turned into a byte array using UTF-8 encoding.
+    The bytes are then sent to the output Kafka topic.
+
+    Custom error handling for the whole pipeline is implemented here.
+    """
+
+    def publish(self, input_msg: dict[str, Any], report: str | bytes) -> None:
+        """Publish an EOL-terminated JSON message to the output Kafka topic.
```

Comment:
why does it need to be EOL terminated?

Reply:
No need at all, inherited from the external data pipeline one. I'd remove it